### DISCUSSION TITLE
EsVesrion and env config should be passed in more build pass.

### DISCUSSION THREAD

**Discussion by magic-akari** [swc-project/swc#4886] [General]
For swc_ecma_codegen we got EsVersion.
But it might be better to have an env config.
we will know how to print the number literial to solve the #4870 .

For swc_ecma_transforms_module.
There is a lot of generated template code, which is not converted from source code.
Optimal code output requires awareness of EsVersion target and env config.

**Comment by kdy1** [comment]
I think it's fine to take them as an arguement

**Comment by Austaras** [comment]
For #4870 babel have a [dedicated pass](https://babeljs.io/docs/en/babel-plugin-transform-literals)

**Comment by Austaras** [comment]
Also is it possible to run module pass before compat?

**Reply by kdy1** [reply]
I think so, and it would allow using some features in modules pass I guess?

**Reply by magic-akari** [reply]
I simply tried it. The first issue that needs to be solved is the inserted swc helpers.
